pkg/ruler/api.go

```diff
@@ -145,8 +171,27 @@ func (a *API) PrometheusRules(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if err := req.ParseForm(); err != nil {
+		level.Error(logger).Log("msg", "error parsing form/query params", "err", err)
+		respondClientError(logger, w, err.Error())
```

Comment:
Are you sure `req.ParseForm()`'s error translate to client error? From quick look of the `req.ParseForm()`'s implementation, it doesn't look like client error is the right choice here.

Reply:
Changed this to respondBadRequest() as that is more appropriate